### PR TITLE
Add gittuf workflows

### PR DESCRIPTION
**Pull request by adityasaky**
Add GHA workflows for gittuf dogfooding.

### CODE REVIEW

File: .github/workflows/gittuf-rsl-main.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+          KEY: ${{ secrets.KEY }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git config --global commit.gpgsign true       # Sign all commits
+          git config --global gpg.x509.program gitsign  # Use gitsign for signing
+          git config --global gpg.format x509           # gitsign expects x509 args
+          git config --global user.name "${{ github.workflow }}"
+          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+
+          echo "$KEY" > /tmp/key
```
Comment (trishankatdatadog) [inline .github/workflows/gittuf-rsl-main.yml:36]: Try parameterising the key name, too, pls.

File: .github/workflows/gittuf-verify.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install gittuf
+        uses: gittuf/gittuf-installer@8ce8fbd070477ec72678151e82b82c3d866d9fcd
+        with:
+          gittuf-version: main
+      - name: Checkout and verify repository
+        run: |
+          gittuf clone https://github.com/${{ github.repository }}
+          cd attestation-verifier
```
[the remark this answers is not in the document]
Comment (adityasaky) [reply .github/workflows/gittuf-verify.yml:19]: github does not, unfortunately, provide that as a parameter. github.repository is `owner/repo` and they have a separate parameter for the owner.

File: .github/workflows/gittuf-rsl-main.yml (hunk 1 above)
Comment (adityasaky) [reply .github/workflows/gittuf-rsl-main.yml:36]: this is currently read in from a secret parametrized in `env`. It's set at the repository level.

**Review by marcelamelara — APPROVED** [review]
LGTM, thanks!

File: .github/workflows/gittuf-rsl-main.yml (hunk 1 above)
Comment (trishankatdatadog) [reply .github/workflows/gittuf-rsl-main.yml:36]: I meant the key file name

File: .github/workflows/gittuf-verify.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (trishankatdatadog) [reply .github/workflows/gittuf-verify.yml:19]: I guess we could strip the owner from `github.repository`, but OK, this works for now.

**Review by trishankatdatadog — APPROVED** [review]

**Pull request merged**